src/routes/settings.go

```diff
@@ -448,7 +450,32 @@ func SettingsRoutes(router *gin.Engine, title string, database *db.Database, _bl
 		c.SecureJSON(http.StatusOK, gin.H{"status": "success", "exportPath": exportDB})
 	})
 	router.POST("/settings/database/importSnapshot", func(c *gin.Context) {
-		importPath := host.GetDataDir() + "yourplace.db.snapshot"
+		dataDir := host.GetDataDir()
+		pattern := filepath.Join(dataDir, "snapshotsyourplacelast*")
+		matches, err := filepath.Glob(pattern)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"status": "Error searching for snapshot files"})
+			return
+		}
+		if len(matches) == 0 {
+			c.AbortWithStatusJSON(http.StatusNotFound, gin.H{"status": "No snapshot file found with prefix 'yourplacelast'"})
```

Comment:
maybe just do a bad request. 404's are really for page not found. not file on system not found